projects/plugins/crm/tests/php/automation/contacts/class-contact-condition-test.php

```diff
@@ -148,4 +162,112 @@ public function test_transitional_status() {
 		$this->assertFalse( $contact_transitional_status_condition->condition_met() );
 	}
 
+	/**
+	 * @testdox Test contact tag added condition.
+	 */
+	public function test_contact_tag_added() {
+		$contact_tag_condition = $this->get_contact_tag_condition( 'tag_added', 'Tag Added' );
+		$contact_data_type     = $this->automation_faker->contact_data( true );
+		$contact_data          = $contact_data_type->get_entity();
+
+		// Create a previous state of a contact.
+		$previous_contact = $contact_data;
+
+		// Testing when the condition has been not been met because the contact does not have said tag.
+		$contact_tag_condition->execute( $contact_data, $previous_contact );
+		$this->assertFalse( $contact_tag_condition->condition_met() );
+
+		// Testing when the condition has been met.
+		$contact_data['tags'][] = array(
+			'id'          => 3,
+			'objtype'     => 1,
```

Comment:
Do we have to use magic numbers here and below?

Reply:
Removed in 446e3fe.